server/src/services/notification.service.ts

```diff
@@ -70,6 +83,107 @@ export class NotificationService {
     return JobStatus.SUCCESS;
   }
 
+  async _getAlbumThumbnailAttachement(album: AlbumEntity): Promise<EmailImageAttachement | undefined> {
+    if (!album.albumThumbnailAssetId) {
+      return;
+    }
+
+    const albumThumbnail = await this.assetRepository.getById(album.albumThumbnailAssetId);
+    if (!albumThumbnail || !albumThumbnail.thumbnailPath) {
+      return;
+    }
+
+    return {
+      filename: 'album-thumbnail.jpg',
+      path: albumThumbnail.thumbnailPath,
+      cid: 'album-thumbnail',
+    };
+  }
+
+  async handleAlbumInvite({ id, recipientId }: INotifyAlbumInviteJob) {
+    const album = await this.albumRepository.getById(id, { withAssets: false });
+    if (!album) {
+      return JobStatus.SKIPPED;
+    }
+
+    const recipient = await this.userRepository.get(recipientId, { withDeleted: false });
+    if (!recipient) {
+      return JobStatus.SKIPPED;
+    }
+
+    const attachement = await this._getAlbumThumbnailAttachement(album);
+
+    const { server } = await this.configCore.getConfig();
+    const { html, text } = this.notificationRepository.renderEmail({
+      template: EmailTemplate.ALBUM_INVITE,
+      data: {
+        baseUrl: server.externalDomain || 'http://localhost:2283',
+        albumId: album.id,
+        albumName: album.albumName,
+        senderName: album.owner.name,
+        recipientName: recipient.name,
+        cid: attachement ? attachement.cid : undefined,
+      },
+    });
+
+    await this.jobRepository.queue({
+      name: JobName.SEND_EMAIL,
+      data: {
+        to: recipient.email,
+        subject: `You have been added to a shared album - ${album.albumName}`,
+        html,
+        text,
+        imageAttachements: attachement ? [attachement] : undefined,
+      },
+    });
+
+    return JobStatus.SUCCESS;
+  }
+
+  async handleAlbumUpdate({ id, senderId }: INotifyAlbumUpdateJob) {
```

Comment:
Should we have a setting to disable this one? I can imagine it potentially getting pretty spammy.

Reply:
It's not a bad idea to add user notifications at some point which lets users (individually) turn them on/off for each event. It might not be a bad idea to implement that before turning on album notifications by default.

Reply:
Yea probably good to wait until then